### ISSUE TITLE
Conditionally run npm goal

### ISSUE THREAD

**Issue by Springrbua** [eirslett/frontend-maven-plugin#869]
**Do you want to request a _feature_ or report a _bug_?**
Request a feature.

**What is the current behavior?**
I am using the `npm` goal to execute `npm ci`, which reinstalls all `node_modules`.
This goal is executed in every build, even if none of the npm dependencies changed.

**What is the expected behavior?**
The npm-command should only run, if the npm dependencies changed.
In our current (non-maven) build, we do that by comparing the modification date of `package-lock.json` with the creation date of `node_modules`. If `node_modules` is older a reinstall is required and `npm ci` is executed.
A flag to enable/disable this behavior might be useful.

**Please mention your frontend-maven-plugin and operating system version.**
OS: Windows 10
frontend-maven-plugin: 1.8.0

**Comment by eirslett** [comment]
I don't think this is really necessary - when you run `npm ci` and nothing has changed, it will noop/not download anything. Maybe there's something wrong with your build setup, which makes it re-download things?

**Comment by Springrbua** [comment]
Regarding the [description](https://docs.npmjs.com/cli/ci.html) of `npm ci`, it removes the existing `node_modules` folder before it beginns to install the dependencies, so it will always download everything.
